src/dotnet/commands/dotnet-add/dotnet-add-proj/Program.cs

```diff
@@ -62,7 +62,7 @@ public override int Run(string fileOrDirectory)
 
         private void AddProject(SlnFile slnFile, string projectPath)
         {
-            var projectPathNormalized = PathUtility.GetPathWithBackSlashes(projectPath);
+            var projectPathNormalized = PathUtility.GetPathWithDirectorySeparator(projectPath);
 
             if (slnFile.Projects.Any((p) =>
                     string.Equals(p.FilePath, projectPathNormalized, StringComparison.OrdinalIgnoreCase)))
```

Comment:
The FilePath property is always file system specific. Only when writing to disk we convert to backslashes. Yes this works on Linux because the tests are passing.